assets/src/data/model/datetime/test/status-helper.js

```diff
@@ -0,0 +1,411 @@
+nextWeek.setDate( today.getDate() + 7 );
+nextMonth.setDate( today.getDate() + 30 );
+lastWeek.setDate( today.getDate() - 7 );
+lastMonth.setDate( today.getDate() - 30 );
+oneMonthAgo.setDate( today.getDate() - 32 );
+twoMonthsAgo.setDate( today.getDate() - 62 );
+oneMonthFromNow.setDate( today.getDate() + 32 );
+twoMonthsFromNow.setDate( today.getDate() + 62 );
+
+const activeDateEntity = () => {
+	const dateEntity = getDateEntity();
+	dateEntity.start = new DateTime( lastWeek.toISOString() );
+	dateEntity.end = new DateTime( nextWeek.toISOString() );
+	return dateEntity;
+};
+const expiredDateEntity = () => {
+	const dateEntity = getDateEntity();
+	dateEntity.start = new DateTime( lastMonth.toISOString() );
+	dateEntity.end = new DateTime( lastWeek.toISOString() );
+	return dateEntity;
+};
+const reallyOldDateEntity = () => {
+	const dateEntity = getDateEntity();
+	dateEntity.start = new DateTime( twoMonthsAgo.toISOString() );
+	dateEntity.end = new DateTime( oneMonthAgo.toISOString() );
+	return dateEntity;
+};
+const upcomingDateEntity = () => {
+	const dateEntity = getDateEntity();
+	dateEntity.start = new DateTime( nextWeek.toISOString() );
+	dateEntity.end = new DateTime( nextMonth.toISOString() );
+	return dateEntity;
+};
+const trashedDateEntity = () => {
+	const dateEntity = getDateEntity();
+	dateEntity.deleted = true;
+	return dateEntity;
+};
+const soldOutDateEntity = () => {
+	const dateEntity = upcomingDateEntity();
+	dateEntity.regLimit = 100;
+	dateEntity.sold = 100;
+	return dateEntity;
+};
+const noCapacityDateEntity = () => {
+	const dateEntity = upcomingDateEntity();
+	dateEntity.regLimit = null;
+	return dateEntity;
+};
+
+describe( 'DateEntity Status "is" Conditionals', () => {
+	it( 'throw errors when invalid Date Entity supplied', () => {
+		expect( () => isActive( inValidDateEntity ) ).toThrow( TypeError );
+		expect( () => isExpired( inValidDateEntity ) ).toThrow( TypeError );
+		expect( () => isRecentlyExpired( inValidDateEntity ) ).toThrow( TypeError );
+		expect( () => isSoldOut( inValidDateEntity ) ).toThrow( TypeError );
+		expect( () => isTrashed( inValidDateEntity ) ).toThrow( TypeError );
+		expect( () => isUpcoming( inValidDateEntity ) ).toThrow( TypeError );
+	} );
+	it( 'do not throw errors when a valid Date Entity supplied', () => {
+		const validDateEntity = getDateEntity();
+		expect( () => isActive( validDateEntity ) ).not.toThrow( TypeError );
+		expect( () => isExpired( validDateEntity ) ).not.toThrow( TypeError );
+		expect( () => isRecentlyExpired( validDateEntity ) ).not.toThrow( TypeError );
+		expect( () => isSoldOut( validDateEntity ) ).not.toThrow( TypeError );
+		expect( () => isTrashed( validDateEntity ) ).not.toThrow( TypeError );
+		expect( () => isUpcoming( validDateEntity ) ).not.toThrow( TypeError );
+	} );
+	describe( 'isActive()', () => {
+		it( 'returns false if Date Entity start and end dates are in future',
+			() => expect( isActive( upcomingDateEntity() ) ).toBe( false )
```

Comment:
This returns the value of the `expect` call.  Test callbacks should not return values (see https://github.com/jest-community/eslint-plugin-jest/pull/136).